smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -0,0 +1,123 @@
+package com.eis.smslibrary;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.Peer;
+
+/**
+ * Implementation of {@link Peer} for SMS communication channel
+ *
+ * @author Luca Crema, Marco Mariotto
+ */
+public class SMSPeer implements Peer<String>, Comparable<SMSPeer> {
+
+    public static final int MAX_TELEPHONE_NUMBER_LENGTH = 20;
+    public static final int MIN_TELEPHONE_NUMBER_LENGTH = 3;
+
+    private String telephoneNumber;
+
+    /**
+     * @param telephoneNumber a valid telephone number (checkTelephoneNumber state must be TELEPHONE_NUMBER_VALID)
+     */
+    public SMSPeer(String telephoneNumber) {
+        this.telephoneNumber = telephoneNumber;
```

Comment:
One thing we do is check if the phone number is valid when it gets created, so if SMSPeer doesn't throw exceptions when it's created you instantly know that the phone number is valid, making every call to SMSPeer.checkState() redundant.

Reply:
But you are forcing the initialization of a variable that won't be used because it has thrown an exception. [As I said earlier](https://github.com/Cogno-IDU/smslibrary/pull/4#discussion_r349953436) I think we should leave the `static checkPhoneNumber(...)` and we could add the exception throwing in case of failed check.